packages/signals/signals/src/core/signals/signals.ts

```diff
@@ -83,6 +83,12 @@ export class Signals implements ISignals {
    * - Registers custom signal generators.
    */
   async start(analytics: AnyAnalytics): Promise<void> {
+    if (
+      !analytics.settings.cdnSettings.autoInstrumentationSettings ||
+      Math.random() >
+        analytics.settings.cdnSettings.autoInstrumentationSettings.sampleRate
```

Comment:
Hey, I don't understand this Math.random() thing, maybe you could explain? I assumed we would just do:
`analytics.settings.cdnSettings.autoInstrumentationSettings.sampleRate === 0`?

Reply:
It's a sample rate from [0,1] that we "roll against" to see if we should send signals or not (not just a boolean in the settings)

Reply:
If the sample rate is 0, we just want to stop from sending signals -- we don't want to stop processing them, right?
This will stop the entire signals initialization. So this really only implies to the signal collection step?